duckduckgo/iOS
# support keyboard shortcuts

Task/Issue URL: https://app.asana.com/0/inbox/392891325557408/1108655038447238/1108820695682454
Tech Design URL:
CC:

**Description**:

Adds keyboard shortcuts to make iPad + keyboard users happy.

* ⌘T — Open new tab
* ⌘W — Close tab
* ⌘N — Open split view
* ⇧⌘] and ⇧⌘[ or Ctrl-Tab and Ctrl-⇧-Tab — Next/Previous Tab
* Ctrl-⇧-\ — Show all tabs
* Esc - cancel typing in the address bar OR close the tab switcher OR close the autocomplete
* Enter - close the tab switcher
* ⌘[ and ⌘] or ⌘→ and ⌘← — Back/forward
* Alt-⌘-F — Edit text field
* ↑↓ — Navigate through auto complete suggestions

These work once the web view has become the first responder.  An up or down arrow press will make it the first responder.  If we specify "space" as a keyboard command then it interferes with the address / search field text entry:

* Space bar — Scroll down a page (⇧-Space reverses the direction).
* ←↑→↓ — Scroll web page (← and → work when zoomed in)
* ⌘↑ and ⌘↓ — Scroll to beginning/end of page

In addition we're enabling the plus button for domains for parity with Android.

**Steps to test this PR**:

1. In the simulator go to Hardware -> Keyboard and check "Send Keyboard Shortcuts to Device"
1. Try out all of the shortcuts listed above on phone (with centered search screen variant) and iPad 
1. Check all auto complete suggestions have the plus button and it works as expected

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
